tests/Utilities/MakeWithRandomValues.hpp

```diff
@@ -176,23 +176,62 @@ ReturnType make_with_random_values(
 template <typename T, typename UniformRandomBitGenerator,
           typename RandomNumberDistribution,
           Requires<cpp17::is_floating_point_v<tt::get_fundamental_type_t<T>>> =
-              nullptr>
+              nullptr,
+          Requires<tt::is_iterable_v<T>> = nullptr>
 T make_with_random_values(
     const gsl::not_null<UniformRandomBitGenerator*> generator,
     const gsl::not_null<RandomNumberDistribution*> distribution) noexcept {
-  T result{std::numeric_limits<tt::get_fundamental_type_t<T>>::signaling_NaN()};
+  T result{};
+  // T is a container type, so we fill with NaNs first
```

Comment:
It's really only useful to ensure that all values are truly overwritten. If that's trustworthy enough, I'm happy to remove all of the initializations for these convenience functions.

Reply:
`fill_with_random_values` is a public interface, so I feel like not trusting it here (in the same file where it is defined) is strange because we do trust it everywhere else.  On the other hand, this is a testing utility, so some amount of redundant checking is fine.  I'll leave this one up to you or anyone else who has a preference.

Reply:
Yea, we should trust fill_with_random_values. Plus I don't think it actually fills anything with `NaN` because you defaulted constructed the `T`

Reply:
I'll remove the initializations, then. In these cases where `used_for_size` is not passed in, the container type is anticipated to be fixed-size type like complex or array, so default construction should provide an object with nonzero size to fill with random values.